straxen/scada.py

```diff
@@ -0,0 +1,246 @@
+import urllib
+import requests
+import pandas as pd
+import numba
+import numpy as np
+
+import strax
+import straxen
+
+try:
+    # Getting secrets:
+    SCData_URL = straxen.get_secret('SCData_URL')
+    SCLastValue_URL = straxen.get_secret('SCLastValue_URL')
+    SCADA_SECRETS = straxen.get_secret('SCADA_SECRETS')
+except ValueError:
+    # We cannot import scada (presumably because we don't have the right
+    # xenon_sectets.py file.
+    SCData_URL = SCLastValue_URL = SCADA_SECRETS = None
+
+export, __all__ = strax.exporter()
+
+
+@export
+def convert_time_zone(df, tz):
+    """
+    Function which converts the current time zone of a given
+    pd.DataFrame into another timezone.
+
+    :param df: pandas.DataFrame containing the Data. Index must be a
+        datetime object with time zone information.
+    :param tz: str representing the timezone the index should be
+        converted to. See the notes for more information.
+    :return: pandas.DataFrame with converted time index.
+
+    Notes:
+        1. ) The input pandas.DataFrame must be indexed via datetime
+        objects which are timezone aware.
+
+        2.)  You can find a complete list of available timezones via:
+        ```
+        import pytz
+        pytz.all_timezones
+        ```
+        You can also specify 'strax' as timezone which will convert the
+        time index into a 'strax time' equivalent.
+        The default timezone of strax is UTC.
+    """
+    if tz == 'strax':
+        df = df.tz_convert(tz='UTC')
+        df.index = df.index.astype(np.int64)
+        df.index.rename(f'time strax', inplace=True)
+    else:
+        df = df.tz_convert(tz=tz)
+        df.index.rename(f'time {tz}', inplace=True)
+    return df
+
+
+def find_scada_parameter():
+    # TODO: Add function which returns SCADA sensor names by short Name
+    raise NotImplementedError
+
+
+@export
+def get_scada_values(parameters,
```

Comment:
Perhaps you thought about it but would a Scada class not make somewhat sense?

Reply:
Yes I thought about this, but I think there is not much of a benefit here. It would allow us to  move line 10 - 18 into the init. I am fine with both.

Reply:
Perhaps as the code gets more complex, certain things are nice to set in a class (like the tz on the init). Perhaps it's easier to already set it up now such that the code does not change all the time. Perhaps the advantage is limited now but may end up significant later.